### PR TITLE
[NativeAOT-LLVM] Zero-init locals on the shadow stack, simplify lowering code, allow STRUCT-typed PHIs

### PR DESCRIPTION
**Pull request by SingleAccretion**
A few changes enhancing our support for locals:
1) Allow struct PHIs (simple fix to use the right `getLlvmTypeFor*` method).
2) Zero-initialize locals on the shadow stack.
3) Do not create no-op `ADD(SS, 0)` nodes: centralize and simplify code.
4) Finally, fix the layout dumping code to not print locals not on the shadow stack as `[SS+00H]`, as well as print the total size of the shadow frame.

### CODE REVIEW

**Comment by SingleAccretion** [comment]
@yowl

File: src/coreclr/jit/llvmcodegen.cpp [hunk 1]
```diff
@@ -1303,6 +1307,13 @@ void Llvm::buildReturn(GenTree* node)
     }
     else
     {
+        retValValue = getGenTreeValue(retValNode);
```
Comment (yowl) [inline src/coreclr/jit/llvmcodegen.cpp]: we are reusing `retValValue` below, nothing incorrect about that, just wanted to ask if it was deliberate and just saves a variable?

Comment (SingleAccretion) [reply src/coreclr/jit/llvmcodegen.cpp]: Yes, it is; the code is temporary and will go away in #2098 so I didn't feel it needed more than this.

**Comment by SingleAccretion** [comment]
Windows x64 Debug threading failure looks to be intermittent.

Windows x64 Release failure is persistent (across at least 2 runs), and rather curious:
```
2022-11-20T01:47:48.8585828Z ILLink : error IL1039: Reference assembly 'D:\a\_work\1\s\artifacts\bin\testaotc\net7.0-windows-Release-x64\framework\comhost.dll' could not be loaded. [D:\a\_work\1\s\src\libraries\System.Threading\tests\System.Threading.Tests.csproj]
2022-11-20T01:47:48.8690283Z ILLink : error IL1039: Reference assembly 'D:\a\_work\1\s\artifacts\bin\testaotc\net7.0-windows-Release-x64\framework\hostfxr.dll' could not be loaded. [D:\a\_work\1\s\src\libraries\System.Threading\tests\System.Threading.Tests.csproj]
2022-11-20T01:47:48.8706430Z ILLink : error IL1039: Reference assembly 'D:\a\_work\1\s\artifacts\bin\testaotc\net7.0-windows-Release-x64\framework\hostpolicy.dll' could not be loaded. [D:\a\_work\1\s\src\libraries\System.Threading\tests\System.Threading.Tests.csproj]
2022-11-20T01:47:48.8725074Z ILLink : error IL1039: Reference assembly 'D:\a\_work\1\s\artifacts\bin\testaotc\net7.0-windows-Release-x64\framework\ijwhost.dll' could not be loaded. [D:\a\_work\1\s\src\libraries\System.Threading\tests\System.Threading.Tests.csproj]
2022-11-20T01:47:48.8746976Z ILLink : error IL1039: Reference assembly 'D:\a\_work\1\s\artifacts\bin\testaotc\net7.0-windows-Release-x64\framework\Microsoft.DiaSymReader.Native.amd64.dll' could not be loaded. [D:\a\_work\1\s\src\libraries\System.Threading\tests\System.Threading.Tests.csproj]
2022-11-20T01:47:48.8753900Z ILLink : error IL1039: Reference assembly 'D:\a\_work\1\s\artifacts\bin\testaotc\net7.0-windows-Release-x64\framework\nethost.dll' could not be loaded. [D:\a\_work\1\s\src\libraries\System.Threading\tests\System.Threading.Tests.csproj]
2022-11-20T01:47:48.8769139Z D:\a\_work\1\s\.dotnet\sdk\6.0.100\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.ILLink.targets(109,5): error NETSDK1144: Optimizing assemblies for size failed. Optimization can be disabled by setting the PublishTrimmed property to false. [D:\a\_work\1\s\src\libraries\System.Threading\tests\System.Threading.Tests.csproj]
2022-11-20T01:47:48.8775706Z     0 Warning(s)
2022-11-20T01:47:48.8777639Z     7 Error(s)
```
As if something is passing unmanaged `.dll`s to the linker or (more likely) failing to filter them out... Unclear how this could be related to these WASM changes. Investigating.

Edit: doesn't immediately reproduce locally, let's see what another CI run says.

Edit: the CI's answer is definitive, so more digging is required.

**Comment by SingleAccretion** [comment]
I was not able to root-cause the CI error after all, but it should now be gone.

**Comment by yowl** [comment]
Looks good, thanks!

**Pull request merged**